mobius-extras/src/test/java/com/spotify/mobius/extras/ToggledEventSourceTest.java

```diff
@@ -0,0 +1,158 @@
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.extras;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import com.spotify.mobius.disposables.Disposable;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ToggledEventSourceTest {
+
+  private FakeDisposingEventSource<String> dataEventSource;
+  private ToggledEventSource<String> underTest;
+  private QueueingEventSubject<Boolean> togglingEventSource;
+
+  @Before
+  public void setUp() {
+    dataEventSource = new FakeDisposingEventSource<>();
+    togglingEventSource = new QueueingEventSubject<>(10);
+    underTest =
+        (ToggledEventSource<String>)
+            ToggledEventSource.from(dataEventSource, togglingEventSource, false);
+  }
+
+  @Test
+  public void testThatActiveEventSourceInitializesWithoutSubscribingToBase() {
+    togglingEventSource.accept(true);
+    dataEventSource.assertConsumerCount(0);
+  }
+
+  @Test
+  public void testActiveEventSourceWithoutSubscribersResubscribesToBaseWhenSubscriberAdded() {
+    togglingEventSource.accept(true);
+    underTest.subscribe(value -> {});
+
+    dataEventSource.assertConsumerCount(1);
+  }
+
+  @Test
+  public void testActiveEventSourceUnsubscribesFromBaseWhenLastSubscriberRemoved() {
+    final List<String> received = new ArrayList<>(1);
+
+    togglingEventSource.accept(true);
+    underTest.subscribe(received::add).dispose();
+
+    dataEventSource.assertConsumerCount(0);
+    assertThat(received.size(), equalTo(0));
+  }
+
+  @Test
+  public void testFilterForwardsEventsWhenActive() {
+    final List<String> received = new ArrayList<>(1);
+    togglingEventSource.accept(true);
+    underTest.subscribe(received::add);
+
+    dataEventSource.emit("a");
+
+    dataEventSource.assertConsumerCount(1);
+    assertThat(received.size(), equalTo(1));
+    assertThat(received.get(0), equalTo("a"));
+  }
+
+  @Test
+  public void testToggledEventSourceBlocksEventsAndRemovesConsumersWhenInactive() {
+    final List<String> received = new ArrayList<>(1);
+    togglingEventSource.accept(true);
+    underTest.subscribe(received::add);
+    togglingEventSource.accept(false);
+    dataEventSource.emit("b");
+
+    dataEventSource.assertConsumerCount(0);
+    assertThat(received.size(), equalTo(0));
```

Comment:
and here (with an import of the right `assertThat`), you could write:
```
assertThat(received).isEmpty();
```
it's a matter of taste, but I quite like AssertJ - and Truth even better, but we're currently using AssertJ in Mobius.